### PR TITLE
Surface Bundler error to browser

### PR DESCRIPTION
**Pull request by mohsen1**
Instead of sending a generic error with the 500 response, print the error message and stack trace in the response HTML payload. 

Feel free to comment. 

screen shot 2018-05-31 at 4 23 03 am

### CODE REVIEW

**Comment by DeMoorJasper** [comment]
Seems like an awesome idea, as we already have the overlay

**Comment by mohsen1** [comment]
Currently I'm stuck with pretty printing output of babel-code-frame in HTML. I know that https://new.babeljs.io/repl is doing something similar (without syntax highlighting)but I'm not sure how they print the code frame. I need to look into it. 


Should I consolidate this code with the overlay? Overlay is manually putting together the HTML as well. Should we depend on a templating library for generating the HTML?

**Comment by DeMoorJasper** [comment]
@mohsen1 I think it’s safe to write a general function that generates the html and just pass a boolean or something that rewrites the css very slightly to be overlay or static

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

File: src/Bundler.js [hunk 1]
```diff
@@ -713,7 +713,11 @@ class Bundler extends EventEmitter {
 
   async serve(port = 1234, https = false) {
     this.server = await Server.serve(this, port, https);
-    this.bundle();
```
Comment (mohsen1) [inline src/Bundler.js]: Previously if this line was throwing an error we had a silent unhandled promise rejection

**Comment by mohsen1** [comment]
@DeMoorJasper @devongovett Can you guys take a look? This is working :)

File: src/Server.js [hunk 2]
```diff
@@ -71,10 +75,28 @@ function middleware(bundler) {
...
 
-    function send500() {
-      res.setHeader('Content-Type', 'text/plain; charset=utf-8');
+    function send500(error) {
+      res.setHeader('Content-Type', 'text/html; charset=utf-8');
       res.writeHead(500);
-      res.end('🚨 Build error, check the console for details.');
+      let errorMesssge = '<h1>🚨 Build Error</h1>';
+      if (process.env.NODE_ENV === 'production') {
+        errorMesssge += '<p><b>Check the console for details.</b></p>';
```
Comment (DeMoorJasper) [inline src/Server.js]: Not sure if this is necessary, the built-in parcel server should never be used in production

File: src/Server.js [hunk 3]
```diff
@@ -71,10 +75,28 @@ function middleware(bundler) {
...
+      } else {
+        const {message, stack} = prettyError(error, {color: true});
+        errorMesssge += `<p><b>${message}</b></p>`;
+        if (stack) {
+          errorMesssge += `<div style="background: black; padding: 1rem;">${ansiToHtml.toHtml(
+            stack
+          )}</div>`;
+        }
+      }
+      res.end(
```
Comment (DeMoorJasper) [inline src/Server.js]: Why not put use one template string here instead of an array?

File: src/Server.js (hunk 2 above)
Comment (DeMoorJasper) [reply src/Server.js]: Parcel can be used as middleware in a production server, so I guess it's good to check this though.

File: src/Server.js (hunk 3 above)
Comment (mohsen1) [reply src/Server.js]: This is to make sure we're not sending extra whitespaces or not ending up with awkward code indentations. Depending on project coding style I can update.

File: src/Server.js (hunk 2 above)
Comment (mohsen1) [reply src/Server.js]: Yeah it should never happen but just in case...

File: src/Server.js (hunk 3 above)
Comment (DeMoorJasper) [reply src/Server.js]: Ah, alright, I don't think the weird indentation matters, but that's personal taste.

**Comment by DeMoorJasper** [comment]
Looks good to me, I'm just not 100% convinced and it's a pretty large change so I'll let someone else review it

File: test/server.js [hunk 4]
```diff
@@ -23,14 +23,13 @@ describe('server', function() {
...
         res => {
-          if (res.statusCode !== 200) {
-            return reject(new Error('Request failed: ' + res.statusCode));
-          }
-
           res.setEncoding('utf8');
           let data = '';
           res.on('data', c => (data += c));
           res.on('end', () => {
+            if (Math.floor(res.statusCode / 100) !== 2) {
```
Comment (DeMoorJasper) [inline test/server.js]: Does this ever respond with a code not equal to 200?

Comment (mohsen1) [reply test/server.js]: Yes. My new test case has 500 response

Comment (mohsen1) [reply test/server.js]: if you ever respond with 201 or other 2xx you'll not need to update this line. I'm open to reverting back this to comparing strictly against 200 if you feel strongly about it

**Comment by mohsen1** [comment]
@devongovett can you take a look? Thanks!

**Pull request merged**